clients/catalyst.go

```diff
@@ -70,15 +77,30 @@ func (c *catalyst) UploadVOD(ctx context.Context, upload UploadVODRequest) error
 	if err != nil {
 		return err
 	}
-	var res json.RawMessage
-	err = c.DoRequest(ctx, Request{
-		Method:      "POST",
-		URL:         "/api/vod",
-		Body:        bytes.NewReader(body),
-		ContentType: "application/json",
-	}, &res)
-	glog.Infof("Catalyst upload VOD request: req=%q err=%q res=%q", body, err, string(res))
-	return err
+	for attempt := 1; ; attempt++ {
+		var res json.RawMessage
+		err = c.DoRequest(ctx, Request{
+			Method:      "POST",
+			URL:         "/api/vod",
+			Body:        bytes.NewReader(body),
+			ContentType: "application/json",
+		}, &res)
+		glog.Infof("Catalyst upload VOD request: req=%q err=%q res=%q", body, err, string(res))
+
+		if !isTooManyRequestsErr(err) {
+			return err
+		}
+
+		if attempt >= maxAttempts {
```

Comment:
So are we giving up if we get rate limited more than 3 times?

Reply:
no! we will retry asynchronously instead through a delayed RabbitMQ event. Check the logic on `upload.go`